scripts/set-env-variable/on-start.sh

```diff
@@ -9,11 +9,24 @@ set -e
 #   1. Ensure the Notebook Instance execution role has permission of SageMaker:ListTags
 #
 
+#sudo -u ec2-user -i <<'EOF'
+
 # PARAMETERS
-YOUR_ENV_VARIABLE_NAME=your_env_variable_name
+YOUR_ENV_VARIABLE_NAME=<ENV_VAR_NAME>
 
 NOTEBOOK_ARN=$(jq '.ResourceArn' /opt/ml/metadata/resource-metadata.json --raw-output)
 TAG=$(aws sagemaker list-tags --resource-arn $NOTEBOOK_ARN  | jq -r --arg YOUR_ENV_VARIABLE_NAME "$YOUR_ENV_VARIABLE_NAME" .'Tags[] | select(.Key == $YOUR_ENV_VARIABLE_NAME).Value' --raw-output)
 touch /etc/profile.d/jupyter-env.sh
 echo "export $YOUR_ENV_VARIABLE_NAME=$TAG" >> /etc/profile.d/jupyter-env.sh
-initctl restart jupyter-server --no-wait
+
+# restart command is dependent on current running Amazon Linux and JupyterLab
+CURR_VERSION_AL=$(cat /etc/system-release)
+CURR_VERSION_JS=$(jupyter --version)
+
+if [[ $CURR_VERSION_JS == *$"jupyter_core     : 4.9.1"* ]] && [[ $CURR_VERSION_AL == *$" release 2018"* ]]; then
+	sudo initctl restart jupyter-server --no-wait
+else
+	sudo systemctl --no-block restart jupyter-server.service || true
+fi
```

Comment:
I have a different opinion about how this should be implemented. I think a custom shell script should be baked into the AMIs, which will handle the restart. Then the individual LCC scripts will just need to execute that baked-in script.

That way we don't need to deal with the kernel and jupyter version detection here, which is error prone. e.g. Imagine that one day the `/etc/system-release` content has changed, or the `jupyter --version` returns something else, then this logic here will be broken.